### PR TITLE
document cylc rose

### PR DESCRIPTION
**Pull request by oliver-sanders**
Sibling: https://github.com/cylc/cylc-rose/pull/22

Document the Cylc Rose plugin within the Cylc docs.

Options considered:

* Document in Rose (`metomi.github.io/rose`)
  * Poor visibility.
  * Cylc Rose is more of a plugin to Cylc than it is to Rose.
* Document in-place (`cylc.org/cylc-rose`)
  * Poor visibility, won't appear in search results.
  * Nice and simple.
* Document in Cylc (`cylc.org/cylc-doc`)
  * Good visibility.
  * There will be other Cylc plugins to document alongside.

Decided to try and document from the cylc.rose package docstring within cylc-doc.

Contains one un-related change to add a Sphinx reference label to the install docs so they can be easily referenced from other projects via intersphinx.

### CODE REVIEW

**Review by MetRonnie — APPROVED** [review]
Looks like `setup.py` in cylc-rose will need changing to use `cylc-flow>8.0a2`?
> ERROR: Could not find a version that satisfies the requirement cylc-flow>=8.0a3 (from cylc-rose[all])

**Comment by oliver-sanders** [comment]
Done in cylc/cylc-rose#22 (though yeah we will need to change that to b0 when the time comes)

**Review by wxtim — APPROVED** [review]
Happy to approve. 
Should we put off merge until we actually have a cylc-rose release to prevent breaking cylc-doc on master?

**Comment by oliver-sanders** [comment]
Right, `pip install` was hanging, something about the Rose alpha 1 on pypi is making it unhappy so I've changed things to install from rose@master for tests and rose@<tag> for deployment.

The Rose version doesn't really make a difference to the docs, it just about satisfying the imports of cylc-rose so we can document from the code.

Could someone give this a quick once-over and whack the button.

Note the tutorial suite is failing *again*, looks like the wind field is missing from datapoint.

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -24,8 +24,14 @@ name: deploy
...
-      tag:
-        description: 'Tag present on both cylc-flow and cylc-docs.'
+      cylc-flow-tag:
+        description: 'Cylc Flow GitHub release tag (must be present in cylc-docs)'
+        required: true
+      cylc-rose-tag:
+        description: 'Cylc Rose GitHub release tag'
+        required: true
+      metomi-rose-tag:
+        description: 'Rose GitHub release tag (currently only effects cylc-rose install)'
```
Comment (MetRonnie) [inline .github/workflows/deploy.yml:34]: Minor typo
```suggestion
        description: 'Rose GitHub release tag (currently only affects cylc-rose install)'
```

**Comment by wxtim** [comment]
Confirmed that it's the data point thing. Again.

**Pull request merged**